core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2132,17 +2132,23 @@ class Log(@volatile private var _dir: File,
 
   /**
    * Get all segments beginning with the segment that includes "from" and ending with the segment
-   * that includes up to "to-1" or the end of the log (if to > logEndOffset)
+   * that includes up to "to-1" or the end of the log (if to > logEndOffset).
+   *
    */
   def logSegments(from: Long, to: Long): Iterable[LogSegment] = {
-    lock synchronized {
-      val view = Option(segments.floorKey(from)).map { floor =>
-        if (to < floor)
-          throw new IllegalArgumentException(s"Invalid log segment range: requested segments in $topicPartition " +
-            s"from offset $from mapping to segment with base offset $floor, which is greater than limit offset $to")
-        segments.subMap(floor, to)
-      }.getOrElse(segments.headMap(to))
-      view.values.asScala
+    if (from == to) {
+      // Handle non-segment-aligned empty sets
+      List.empty[LogSegment]
+    } else {
+      lock synchronized {
+        val view = Option(segments.floorKey(from)).map { floor =>
+          if (to < floor)
+            throw new IllegalArgumentException(s"Invalid log segment range: requested segments in $topicPartition " +
```

Comment:
While you are here, it's probably better to consistently throw IllegalArgumentException if `from > to`. Currently, we only throw IllegalArgumentException if we get into this path.

Reply:
Done.